Client/Frontend/Home/HistoryHighlights/FxHomeHistoryHighlightsVM.swift

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+class FxHomeHistoryHightlightsVM {
+
+    // MARK: - Properties & Variables
+    var historyItems = [Tab]()
+    var tabManager: TabManager
+
+    var onTapItem: (() -> Void)?
+
+    private var recentTabs = [Tab]()
+    private var maxItemsAllowed: Int {
+        HistoryHighlightsCollectionCellConstants.maxNumberOfItemsPerColumn * HistoryHighlightsCollectionCellConstants.maxNumberOfColumns
+    }
+
+    // MARK: - Inits
+    init(with tabManager: TabManager) {
+        self.tabManager = tabManager
+    }
+
+    // MARK: - Public methods
+    public func updateData() {
+        loadItems()
+    }
+
+    public func switchTo() {
+        if BrowserViewController.foregroundBVC().urlBar.inOverlayMode {
```

Comment:
Nit: `BrowserViewController.foregroundBVC()` could be passed on dependency injection, to be able to write unit tests of this public function.